### ISSUE TITLE
HTTPRoute.spec.rules.backendRefs.namespace does not work

### ISSUE THREAD

**Issue by ct1150** [linkerd/linkerd2#10902]
### What is the issue?

Hello, I'm using the latest version 2.13 of linkerd with the dynamic routing feature. I'm trying to achieve router-based routing using headers. However, I've encountered an issue: when using header-based forwarding, the namespace field added in backendRefs seems to be ineffective. Even if I use a non-existent namespace or an incorrect namespace, the request still gets forwarded to the service in the local namespace. Could you please help me confirm if this is a bug? Thank you.

### How can it be reproduced?

```text
apiVersion: policy.linkerd.io/v1beta2
kind: HTTPRoute
metadata:
  name: frontend-router
  namespace: test
spec:
  parentRefs:
  - group: core
    kind: Service
    name: frontend-podinfo
    port: 9898
  rules:
  - backendRefs:
    - group: ""
      kind: Service
      name: frontend-podinfo
      namespace: somespace
      port: 9898
      weight: 1
    matches:
    - headers:
      - name: x-request-id
        type: Exact
        value: alternative
      path:
        type: PathPrefix
        value: /
  - backendRefs:
    - group: ""
      kind: Service
      name: frontend-podinfo
      port: 9898
      weight: 1
    matches:
    - path:
        type: PathPrefix
        value: /
```

### Logs, error output, etc

When making a request with headers, it is expected to route to the corresponding service in the specified namespace. Alternatively, when configuring a non-existent namespace, it should return a 500 error. However, in reality, the request always succeeds because it consistently targets services in the local namespace.

### output of `linkerd check -o short`

```text
linkerd-version
---------------
‼ cli is up-to-date
    is running version 2.13.2 but the latest stable version is 2.13.3
    see https://linkerd.io/2.13/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 2.13.2 but the latest stable version is 2.13.3
    see https://linkerd.io/2.13/checks/#l5d-version-control for hints

linkerd-control-plane-proxy
---------------------------
‼ control plane proxies are up-to-date
    some proxies are not running the current version:
        * linkerd-destination-75d49778bf-7qmqh (stable-2.13.2)
        * linkerd-destination-75d49778bf-lqwpd (stable-2.13.2)
        * linkerd-destination-75d49778bf-mml2j (stable-2.13.2)
        * linkerd-identity-549685b8d7-454x2 (stable-2.13.2)
        * linkerd-identity-549685b8d7-8t6wb (stable-2.13.2)
        * linkerd-identity-549685b8d7-svtlh (stable-2.13.2)
        * linkerd-proxy-injector-689788498f-9l6bh (stable-2.13.2)
        * linkerd-proxy-injector-689788498f-s5vnt (stable-2.13.2)
        * linkerd-proxy-injector-689788498f-sdqtz (stable-2.13.2)
    see https://linkerd.io/2.13/checks/#l5d-cp-proxy-version for hints

linkerd-ha-checks
-----------------
‼ pod injection disabled on kube-system
    kube-system namespace needs to have the label config.linkerd.io/admission-webhooks: disabled if injector webhook failure policy is Fail
    see https://linkerd.io/2.13/checks/#l5d-injection-disabled for hints

Status check results are √
```

### Environment

v1.23.17-eks-a59e1f0
eks
amazon linux
2.13.2

### Would you like to work on fixing this bug?

None

**Comment by ct1150** [comment]
I found some description in code:
```text
namespace:
                            description: "Namespace is the namespace of the backend.
                              When unspecified, the local namespace is inferred. \n
                              Note that when a namespace is specified, a ReferenceGrant
                              object is required in the referent namespace to allow
                              that namespace's owner to accept the reference. See
                              the ReferenceGrant documentation for details. \n Support:
                              Core"
```
but there is only the ReferenceGrant documentation in Kubernetes Gateway API;is there a crd by linkerd?

**Comment by alpeb** [comment]
Thanks for the report @ct1150 . This is currently being addressed in #10909. Note that even though ReferenceGrant is mentioned in the CRD doc, it's not relevant for inner-mesh connections; there is nothing to be done in order to allow cross-namespace traffic in the mesh when using HTTPRoutes.

**Comment by ct1150** [comment]
ok，thanks

**Comment by hawkw** [comment]
The fix for this issue (#10909) has been released in [stable-2.13.4](https://github.com/linkerd/linkerd2/releases/tag/stable-2.13.4). I think we can close this as fixed.
